tests/test_integration.py

```diff
@@ -148,11 +148,87 @@ def test_single_stimuli_single_outcome(self):
         self.check_single_stimuli(x1, x2)
         self.check_single_outcome(outcomes)
 
+    def test_single_stimuli_multi_outcome(self):
+        """Test a single-stimulus experiment with a multiple outcomes."""
+        # Read config from .ini file
+        with open('tests/configs/singleStimuli.ini', 'r') as f:
+            dummy_simple_config = f.read()
+        self.setup_request["message"]["config_str"] = dummy_simple_config
+
+        self.s.versioned_handler(self.setup_request)
+
+        x1 = [0.1, 0.2, 0.3, 1, 2, 3, 4]
+        x2 = [4, 0.1, 3, 0.2, 2, 1, 0.3]
+        outcomes = [[1, 0], [-1, 0], [0.1, 0], [0, 0], [-0.1, 0], [0, 0], [0, 0]]
+
+        i = 0
+        while not self.s.strat.finished and i<7:
+            #! TODO: Ask request should be supported!
+            # self.s.unversioned_handler(self.ask_request)
```

Comment:
The ask-request is not working in this case (single stimuli with multiple outcomes). I'm not sure if it's because multiple outcomes are still not supported or if I need to change something in the config file `tests/configs/singleStimuli.ini` 🤔

Here is the full error message of this test case when this code line (ask-request) is uncommented:

```
======================================================================
ERROR: test_single_stimuli_multi_outcome (tests.test_integration.IntegrationTestCase)
Test a single-stimulus experiment with a multiple outcomes.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/missa/projects/aepsych/tests/test_integration.py", line 167, in test_single_stimuli_multi_outcome
    self.s.unversioned_handler(self.ask_request)
  File "/home/missa/projects/aepsych/aepsych/server/server.py", line 467, in unversioned_handler
    ret_val = message_map[type](request)
  File "/home/missa/projects/aepsych/aepsych/server/server.py", line 506, in handle_ask
    new_config = self.ask()
  File "/home/missa/projects/aepsych/aepsych/server/server.py", line 721, in ask
    next_x = self.strat.gen()[0]
  File "/home/missa/projects/aepsych/aepsych/strategy.py", line 460, in gen
    return self._strat.gen(num_points=num_points, **kwargs)
  File "/home/missa/projects/aepsych/aepsych/strategy.py", line 33, in wrapper
    self.fit()
  File "/home/missa/projects/aepsych/aepsych/strategy.py", line 315, in fit
    self.model.fit(self.x, self.y)
  File "/home/missa/projects/aepsych/aepsych/models/gp_classification.py", line 231, in fit
    self._fit_mll(mll, **kwargs)
  File "/home/missa/projects/aepsych/aepsych/models/base.py", line 426, in _fit_mll
    fit_gpytorch_mll(mll, optimizer_kwargs=optimizer_kwargs, **kwargs)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/botorch/fit.py", line 114, in fit_gpytorch_mll
    return dispatcher(
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/botorch/utils/dispatcher.py", line 88, in __call__
    return func(*args, **kwargs)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/botorch/fit.py", line 240, in _fit_fallback
    mll, _ = optimizer(mll, **optimizer_kwargs)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/botorch/optim/fit.py", line 142, in fit_gpytorch_scipy
    res = minimize(
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/scipy/optimize/_minimize.py", line 699, in minimize
    res = _minimize_lbfgsb(fun, x0, args, jac, bounds,
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/scipy/optimize/_lbfgsb_py.py", line 306, in _minimize_lbfgsb
    sf = _prepare_scalar_function(fun, x0, jac=jac, args=args, epsilon=eps,
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/scipy/optimize/_optimize.py", line 263, in _prepare_scalar_function
    sf = ScalarFunction(fun, x0, args, grad, hess,
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/scipy/optimize/_differentiable_functions.py", line 158, in __init__
    self._update_fun()
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/scipy/optimize/_differentiable_functions.py", line 251, in _update_fun
    self._update_fun_impl()
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/scipy/optimize/_differentiable_functions.py", line 155, in update_fun
    self.f = fun_wrapped(self.x)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/scipy/optimize/_differentiable_functions.py", line 137, in fun_wrapped
    fx = fun(np.copy(x), *args)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/scipy/optimize/_optimize.py", line 76, in __call__
    self._compute_if_needed(x, *args)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/scipy/optimize/_optimize.py", line 70, in _compute_if_needed
    fg = self.fun(x, *args)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/botorch/optim/utils.py", line 239, in _scipy_objective_and_grad
    return _handle_numerical_errors(error=e, x=x)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/botorch/optim/utils.py", line 268, in _handle_numerical_errors
    raise error  # pragma: nocover
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/botorch/optim/utils.py", line 237, in _scipy_objective_and_grad
    loss = -mll(*args).sum()
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/gpytorch/module.py", line 30, in __call__
    outputs = self.forward(*inputs, **kwargs)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/gpytorch/mlls/variational_elbo.py", line 77, in forward
    return super().forward(variational_dist_f, target, **kwargs)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/gpytorch/mlls/_approximate_mll.py", line 58, in forward
    log_likelihood = self._log_likelihood_term(approximate_dist_f, target, **kwargs).div(num_batch)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/gpytorch/mlls/variational_elbo.py", line 61, in _log_likelihood_term
    return self.likelihood.expected_log_prob(target, variational_dist_f, **kwargs).sum(-1)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/gpytorch/likelihoods/bernoulli_likelihood.py", line 53, in expected_log_prob
    log_prob = self.quadrature(log_prob_lambda, function_dist)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/gpytorch/utils/quadrature.py", line 81, in forward
    log_probs = func(shifted_locs)
  File "/home/missa/projects/aepsych/mlh_env_ubuntu/lib/python3.8/site-packages/gpytorch/likelihoods/bernoulli_likelihood.py", line 52, in <lambda>
    log_prob_lambda = lambda function_samples: log_normal_cdf(function_samples.mul(observations))
RuntimeError: The size of tensor a (3) must match the size of tensor b (6) at non-singleton dimension 2

----------------------------------------------------------------------
```